src/github/webhook.rs

```diff
@@ -94,28 +94,12 @@ impl fmt::Display for EventName {
     }
 }
 
-#[derive(Debug)]
-pub struct WebhookError(
-    #[allow(dead_code)] // Used in debug display
-    anyhow::Error,
-);
-
-impl From<anyhow::Error> for WebhookError {
-    fn from(e: anyhow::Error) -> WebhookError {
-        WebhookError(e)
-    }
-}
-
 pub fn deserialize_payload<T: serde::de::DeserializeOwned>(v: &str) -> anyhow::Result<T> {
     let mut deserializer = serde_json::Deserializer::from_str(&v);
     let res: Result<T, _> = serde_path_to_error::deserialize(&mut deserializer);
     match res {
         Ok(r) => Ok(r),
-        Err(e) => {
-            tracing::error!("failed to deserialize webhook payload: {v}");
-            let ctx = format!("at {:?}", e.path());
-            Err(e.into_inner()).context(ctx)
-        }
+        Err(e) => Err(anyhow::anyhow!("webhook payload: {v}").context(e)),
```

Comment:
Shouldn't it be the other way around, i.e. return `e` as an error and add the payload as context?

Reply:
I though so to, but I don't think so, if we did that the error chain wouldn't make sense: error -> payload -> error ; while with that it's error -> error -> payload, makes more sense to me (it also renders better IMO).